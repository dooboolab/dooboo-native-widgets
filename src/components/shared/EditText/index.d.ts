import * as React from 'react';
import { ViewStyle, ImageSourcePropType } from 'react-native';

interface Props {
  parentTestID?: string;
  testID?: string;
  errorTestID?: string;
  style?: ViewStyle;
  label?: string;
  textStyle?: TextStyle;
  errorText?: string;
  text?: string;
  placeholder?: string;
  placeholderTextColor?: string;
  secureTextEntry?: boolean;
  onSubmitEditing?: (e: NativeSyntheticEvent<any>) => void;
  onTextChanged?: (text: string) => void;
}

const EditText: React.SFC<Props> = (props) => {

};

export default EditText;
