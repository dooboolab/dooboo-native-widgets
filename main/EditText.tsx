import {Platform, StyleSheet, Text, TextInput, View} from 'react-native';
import React, {FC, useState} from 'react';
import type {StyleProp, TextInputProps, ViewStyle} from 'react-native';

const defaultStyle: ViewStyle = {
  alignSelf: 'stretch',
  flex: 1,

  flexDirection: 'column',
};

const defaultRowStyles = StyleSheet.create({
  container: {
    alignSelf: 'stretch',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    borderBottomWidth: 0.3,
    borderBottomColor: '#666666',
  },
  labelText: {
    fontSize: 14,
    color: '#b9b9c4',
    width: 100,
  },
  input: {
    paddingVertical: 12,
    paddingHorizontal: 8,
    fontSize: 14,
    fontWeight: 'bold',
    flex: 1,
    color: '#2c374e',
  },
  errorText: {
    marginTop: 4,
    fontSize: 12,
    color: '#991111',
  },
});

const defaultColumnStyles = StyleSheet.create({
  container: {
    alignSelf: 'stretch',
    justifyContent: 'space-between',
    borderBottomWidth: 0.3,
    borderBottomColor: '#666666',

    flexDirection: 'column',
  },
  labelText: {
    fontSize: 14,
    color: '#b9b9c4',
  },
  input: {
    paddingVertical: 12,
    fontSize: 14,
    fontWeight: 'bold',
    flex: 1,
    color: '#2c374e',
  },
  errorText: {
    marginTop: 8,
    fontSize: 12,
    color: '#991111',
  },
});

type StylesType = Partial<StyleSheet.NamedStyles<typeof defaultRowStyles>>;

type Props = {
  testID?: TextInputProps['testID'];
  textInputProps?: TextInputProps;
  style?: StyleProp<ViewStyle>;
  styles?: StylesType;
  labelText?: string;
  errorText?: string;
  value?: TextInputProps['value'];
  onChange?: TextInputProps['onChange'];
  onChangeText?: TextInputProps['onChangeText'];
  placeholder?: TextInputProps['placeholder'];
  placeholderTextColor?: TextInputProps['placeholderTextColor'];
  onFocus?: TextInputProps['onFocus'];
  onBlur?: TextInputProps['onBlur'];
  editable?: TextInputProps['editable'];
  autoCapitalize?: TextInputProps['autoCapitalize'];
  secureTextEntry?: TextInputProps['secureTextEntry'];
  onSubmitEditing?: TextInputProps['onSubmitEditing'];
  focusColor?: string;
  errorColor?: string;
  disableColor?: string;
  type?: 'row' | 'column';
};

const EditText: FC<Props> = ({
  testID,
  textInputProps,
  style,
  styles,
  labelText = '',
  errorText = '',
  value = '',
  placeholder,
  placeholderTextColor,
  onChange,
  onChangeText,
  onFocus,
  onBlur,
  autoCapitalize = 'none',
  secureTextEntry = false,
  editable = true,
  focusColor = '#9A77FF',
  errorColor = '#ff7676',
  disableColor = '#999999',
  type = 'column',
}) => {
  const [focused, setFocused] = useState(false);

  const defaultStyles = type === 'row' ? defaultRowStyles : defaultColumnStyles;

  return (
    <View style={[defaultStyle, style]}>
      <View
        style={[
          defaultStyles.container,
          {
            borderColor: errorText
              ? errorColor
              : focused
              ? focusColor
              : disableColor,
            borderBottomColor: errorText
              ? errorColor
              : focused
              ? focusColor
              : disableColor,
          },
        ]}>
        {/* {labelText ? ( */}
        <Text
          style={[
            defaultStyles.labelText,
            styles?.labelText,
            {
              color: errorText
                ? errorColor
                : focused
                ? focusColor
                : disableColor,
            },
          ]}>
          {labelText}
        </Text>
        {/* ) : null} */}
        <TextInput
          {...textInputProps}
          testID={testID}
          autoCapitalize={autoCapitalize}
          secureTextEntry={secureTextEntry}
          style={[
            defaultStyles.input,
            styles?.input,
            // @ts-ignore
            Platform.OS === 'web' && {outlineWidth: 0},
          ]}
          editable={editable}
          onFocus={(e) => {
            setFocused(true);
            onFocus?.(e);
          }}
          onBlur={(e) => {
            setFocused(false);
            onBlur?.(e);
          }}
          value={value}
          placeholder={placeholder}
          placeholderTextColor={placeholderTextColor}
          onChange={onChange}
          onChangeText={onChangeText}
        />
      </View>
      {errorText ? (
        <Text
          style={[
            defaultStyles.errorText,
            styles?.errorText,
            {color: errorColor},
          ]}>
          {errorText}
        </Text>
      ) : null}
    </View>
  );
};

export {EditText};
