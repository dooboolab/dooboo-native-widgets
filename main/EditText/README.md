# EditText

> EditText component is an enhanced version on pure react-native [TextInput] component.</br>

![image](https://user-images.githubusercontent.com/58724686/87130683-4121b100-c2ce-11ea-8663-3f6e7d380a6d.gif)

## Props

|                      | necessary | types                  | default              |
| -------------------- | --------- | ---------------------- | -------------------- |
| labelPosition        |           | 'row'                  |                      |
| placeholder          |           | string                 |         `text`       |
| placeholderTextColor |           | object                 |                      |
| containerStyle       |           | `ViewStyle`            |                      |
| textInputStyle       |           | `ViewStyle`            |                      |
| focusColor           |           | string                 |                      |
| labelText            |           | string                 |                      |
| labelTextStyle       |           | `TextStyle`            |                      |
| value                |           | `TextInputProps`       |                      |
| isErrored            |           | boolean                |        `false`       |
| errorMessage         |           | `React.ReactElement`   |                      |
| errorStyle           |           | `ViewStyle`            |                      |
| numberOfLine         |           |  number                |                      |
| secureTextEntry      |           |  boolean               |                      |
| onFocus              |           |  ()=> {}               |                      |
| onBlur               |           |  ()=> {}               |                      |
| onChangeText         |           |  ()=> {}               |                      |
| onSubmitEditing      |           |  ()=> {}               |                      |

## Installation

```sh
yarn add dooboo-ui
```

## Description
- When you want to use Row type, you should set labelPosition props to 'row'. 
  You can remove the border with containerStyle porps in row type. but In default, you can remove the border with textInputStyle props. 

## Usage
  This is use case for signIn view.

```javascript

import {EditText} from 'dooboo-ui';

const UseCase = ():React.ReactElement => {
  const [email, setEmail] = useState<string>('');
  const [password, setPassword] = useState<string>('');
  const [isEmailErrored, setEmailError] = useState<boolean>(false);
  const [isPswErrored, setPswError] = useState<boolean>(false);

  const onSignIn = (): void => {
    if (email === '' && password === '') {
      setEmailError(true);
      setPswError(true);
    } else if (email === '') {
      setEmailError(true);
    } else if (password === '') {
      setPswError(true);
    }
  };

  return (
    <SafeAreaView>
      <ScrollView
        contentContainerStyle={{
          marginTop: 8,
          alignSelf: 'stretch',
          paddingHorizontal: 20,
          paddingVertical: 100,
        }}
      >
        <View
          style={{
            display: 'flex',
            flexDirection: 'column',
            alignItems: 'center',
            padding: 20,
            paddingTop: 10,
          }}
        >
          <Text
            style={{
              fontWeight: 'bold',
              fontSize: 36,
              lineHeight: 53,
              color: '#1B1B21',
            }}
          >
            SignUp
          </Text>
        </View>
        <View
          style={{
            justifyContent: 'center',
            alignItems: 'center',
          }}
        >
          <EditText
            labelText={'Email'}
            value={email}
            isErrored={isEmailErrored}
            errorStyle={{ borderColor: '#E54E4E' }}
            errorMessage={
              <Text style={{ color: '#E54E4E', marginTop: 8 }}>
                Invaild Email address
              </Text>
            }
            placeholder={'dooboolab@gmail.com'}
            placeholderTextColor={'#707683'}
            onChangeText={(email): void => { setEmail(email); }}
            containerStyle={{
              marginBottom: 30,
              width: 528,
            }}
            labelTextStyle={{
              color: '#323C47',
            }}
            onFocus={(): void => setEmailError(false)}
          />
          <EditText
            labelText={'Password'}
            value={password}
            isErrored={isPswErrored}
            errorStyle={{ borderColor: '#E54E4E' }}
            errorMessage={
              <Text style={{ color: '#E54E4E', marginTop: 8 }}>
                Invaild Password
              </Text>
            }
            placeholder={'********'}
            placeholderTextColor={'#707683'}
            onChangeText={(password): void => { setPassword(password); }}
            secureTextEntry={true}
            containerStyle={{
              width: 528,
            }}
            labelTextStyle={{
              color: '#323C47',
            }}
            onFocus={(): void => setPswError(false)}
          />
        </View>
        <View
          style={{
            alignItems: 'center',
            marginTop: 25,
          }}
        >
          <Text
            style={{
              fontSize: 14,
              lineHeight: 18,
              color: '#231F20',
            }}
          >
            Agree to terms and conditions
          </Text>
        </View>
        <Button
          style={{
            borderWidth: 1,
            borderColor: '#609FFF',
            borderRadius: 6,
            marginTop: 40,
            width: 528,
          }}
          text="이메일 인증하기"
          textStyle={{
            color: '#609FFF',
            fontWeight: 'bold',
            fontSize: 10,
            lineHeight: 15,
            letterSpacing: -0.1,
          }}
          onPress={(): void => onSignIn()}
        />
      </ScrollView>
    </SafeAreaView>
  );
};
```
