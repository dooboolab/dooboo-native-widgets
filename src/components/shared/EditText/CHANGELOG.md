[0.4.4]
- Support boxes [#177](https://github.com/dooboolab/dooboo-ui-native/pull/117).

[0.4.1]
- Support row aligned.

[0.3.0]
- First release.
