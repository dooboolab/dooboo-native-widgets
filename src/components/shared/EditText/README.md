# EditText
> [EditText] component is an enhanced version on pure react-native [TextInput] component.

## Props
|                      | necessary | types        | default                      |
|----------------------|-----------|--------------|------------------------------|
| pararentTestId       |           | string       |                              |
| testID               |           | string       |                              |
| errorTestID          |           | string       |                              |
| style                |           | ViewStyle    |                              |
| label                |           | string       |                              |
| textStyle            |           | TextStyle    |                              |
| errorText            |           | string       |                              |
| text                 |           | string       |                              |
| placeholder          |           | string       |                              |
| placeholderTextColor |           | string       |                              |
| secureTextEntry      |           | boolean      |                              |
| onSubmitEditing      |           | func         |                              |
| onTextChanged        |           | (e) => {}    |                              |

## Getting started

* Import
  ```javascript
  import { EditText } from 'dooboo-native-widgets';
  ```

* Usage
  ```javascript
  function Page(props: Props) {
    const validateEmail = (email: string) => {
      const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
      return re.test(email);
    };

    const fontStyle: TextStyle = {
      fontWeight: 'bold',
      fontSize: 13,
    };

    const [email, setEmail] = useState('');
    const [password, setPassword] = useState('');
    const [errorEmail, setErrorEmail] = useState('');

    const onSignIn = () => {
      if (!validateEmail(email)) {
        setErrorEmail('Not a valid email address');
      } else {
        setErrorEmail('');
      }
    };

    const onTextChanged = (type: string, text: string) => {
      type === 'EMAIL' ? setEmail(text) : setPassword(text);

      if (type === 'EMAIL' && text === '') {
        setErrorEmail('');
      }
    };

    return (
      <StyledScrollView
        contentContainerStyle={{
          marginTop: 8,
          paddingHorizontal: 20,
          paddingBottom: 40,
        }}
      >
        <Container>
          <HeaderTitle>Sign in with Email</HeaderTitle>
          <EditText
            testID='EMAIL_INPUT'
            textStyle={{
              color: '#495057',
            }}
            label='Email'
            placeholder='Write email address'
            placeholderTextColor='#ADB5BD'
            text={ email }
            onTextChanged={ (text: string) => onTextChanged('EMAIL', text) }
            style={ { marginTop: 50 } }
            errorText={errorEmail}
            onSubmitEditing={onSignIn}
          />
          <EditText
            testID='PASSWORD_INPUT'
            textStyle={{
              color: '#ADB5BD',
            }}
            secureTextEntry={true}
            label='Password'
            placeholder='Please write your password'
            placeholderTextColor='#ADB5BD'
            text={ password }
            onTextChanged={ (text: string) => onTextChanged('PASSWORD', text) }
            style={ { marginTop: 36 } }
            onSubmitEditing={onSignIn}
          />
          <StyledSignInButton
            testID='btnEmail'
            onClick={() => onSignIn()}
            textStyle={fontStyle}
            text='Login'
          />
          {/* Email SignUp text */}
          <View
            style={{
              marginTop: 20,
              flexDirection: 'row',
              justifyContent: 'center',
              alignItems: 'center',
            }}
          >
            <StyledText testID='NO_ACCOUNT'>Do not have and account? </StyledText>
            <TouchableOpacity
              onPress={() => null}
              style={{ padding: 4 }}
            >
              <StyledAccentText>Find</StyledAccentText>
            </TouchableOpacity>
          </View>
        </Container>
      </StyledScrollView>
    );
  }
```
