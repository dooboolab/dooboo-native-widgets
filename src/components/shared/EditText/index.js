import React, { useState } from 'react';
import styled from 'styled-components/native';
import PropTypes from 'prop-types';

import {
  StyleSheet,
  Text,
  View,
  TextInput,
  TextStyle,
  ViewPropTypes,
  ViewStyle,
  Platform,
  NativeSyntheticEvent,
} from 'react-native';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'column',
    alignSelf: 'stretch',
    width: '100%',
  },
  underline: {
    borderWidth: 0.6,
    borderColor: '#eaeaf9',
  },
  label: {
    fontSize: 14,
    fontWeight: '500',
    color: '#b9b9c4',
  },
  input: Platform.select({
    ios: {
      paddingTop: 15,
      paddingBottom: 10,
      fontSize: 15,
      fontWeight: '500',
      color: '#2C374E',
    },
    android: {
      paddingLeft: 0,
      paddingBottom: 5,
      fontSize: 15,
      fontWeight: '500',
      color: '#2C374E',
    },
  }),
  errorText: {
    fontSize: 12,
    fontWeight: '500',
    marginTop: 5,
    color: '#ff8989',
  },
});

function EditText(props) {
  const [focused, setFocus] = useState(false);

  return (
    <View
      testID={ props.parentTestID }
      style={[
        styles.container,
        props.style,
      ]}
    >
      {
        props.label
          ? <Text style={[
            styles.label,
            props.errorText
              ? { color: '#FF8989' }
              : focused
                ? { color: '#79B3F5' }
                : null,
          ]}
          >
            { props.label }
          </Text>
          : null
      }
      <TextInput
        testID={props.testID}
        style={[
          styles.input,
          props.textStyle,
        ]}
        autoCapitalize={'none'}
        onFocus={ () => setFocus(true) }
        onBlur={ () => setFocus(false) }
        onSubmitEditing={ props.onSubmitEditing }
        placeholder={ props.placeholder }
        placeholderTextColor={ props.placeholderTextColor }
        value={ props.text }
        onChangeText={ props.onTextChanged}
        secureTextEntry={ props.secureTextEntry }
      >
      </TextInput>
      <View style={[
        styles.underline,
        focused
          ? { borderColor: '#79B3F5' }
          : props.errorText
            ? { borderColor: '#FF8989' }
            : null,
      ]}
      />
      {
        props.errorText
          ? <Text
            testID={props.errorTestID}
            style={styles.errorText}
          >
            {props.errorText}
          </Text>
          : null
      }
    </View>
  );
}

EditText.propTypes = {
  parentTestID: PropTypes.string,
  testID: PropTypes.string,
  errorTestID: PropTypes.string,
  style: ViewPropTypes.style,
  label: PropTypes.string,
  textStyle: Text.propTypes.style,
  errorText: PropTypes.string,
  text: PropTypes.string,
  placeholder: PropTypes.string,
  placeholderTextColor: PropTypes.string,
  secureTextEntry: PropTypes.bool,
  onSubmitEditing: PropTypes.func,
  onTextChanged: PropTypes.func,
};

export default EditText;
