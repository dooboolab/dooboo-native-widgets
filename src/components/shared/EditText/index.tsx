import {
  NativeSyntheticEvent,
  Platform,
  StyleProp,
  StyleSheet,
  Text,
  TextInput,
  TextStyle,
  View,
  ViewStyle,
} from 'react-native';
import React, { useState } from 'react';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'column',
    alignSelf: 'stretch',
    width: '100%',
  },
  underline: {
    borderWidth: 0.6,
    borderColor: '#eaeaf9',
  },
  label: {
    fontSize: 14,
    fontWeight: '500',
    color: '#b9b9c4',
  },
  input: Platform.select({
    ios: {
      paddingTop: 15,
      paddingBottom: 10,
      fontSize: 15,
      fontWeight: '500',
      color: '#2C374E',
    },
    android: {
      paddingLeft: 0,
      paddingBottom: 5,
      fontSize: 15,
      fontWeight: '500',
      color: '#2C374E',
    },
  }),
  errorText: {
    fontSize: 12,
    fontWeight: '500',
    marginTop: 5,
    color: '#ff8989',
  },
});

interface Props {
  parentTestID?: string;
  testID?: string;
  errorTestID?: string;
  style?: StyleProp<ViewStyle>;
  label?: string;
  textStyle?: StyleProp<TextStyle>;
  errorText?: string;
  text?: string;
  placeholder?: string;
  placeholderTextColor?: string;
  secureTextEntry?: boolean;
  onSubmitEditing?: (e: NativeSyntheticEvent<any>) => void;
  onTextChanged?: (text: string) => void;
}

function EditText(props: Props) {
  const [focused, setFocus] = useState(false);

  return (
    <View testID={props.parentTestID} style={[styles.container, props.style]}>
      {props.label ? (
        <Text
          style={[
            styles.label,
            props.errorText
              ? { color: '#FF8989' }
              : focused
                ? { color: '#79B3F5' }
                : null,
          ]}
        >
          {props.label}
        </Text>
      ) : null}
      <TextInput
        testID={props.testID}
        style={[styles.input, props.textStyle]}
        autoCapitalize={'none'}
        onFocus={() => setFocus(true)}
        onBlur={() => setFocus(false)}
        onSubmitEditing={props.onSubmitEditing}
        placeholder={props.placeholder}
        placeholderTextColor={props.placeholderTextColor}
        value={props.text}
        onChangeText={props.onTextChanged}
        secureTextEntry={props.secureTextEntry}
      ></TextInput>
      <View
        style={[
          styles.underline,
          focused
            ? { borderColor: '#79B3F5' }
            : props.errorText
              ? { borderColor: '#FF8989' }
              : null,
        ]}
      />
      {props.errorText ? (
        <Text testID={props.errorTestID} style={styles.errorText}>
          {props.errorText}
        </Text>
      ) : null}
    </View>
  );
}

export default EditText;
