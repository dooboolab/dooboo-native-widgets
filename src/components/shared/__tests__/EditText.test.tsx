import * as React from 'react';

import {
  RenderResult,
  act,
  fireEvent,
  render,
  wait,
  waitForElement,
} from '@testing-library/react-native';

import EditText from '../EditText';
import { View } from 'react-native';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
let component: React.ReactElement;
let testingLib: RenderResult;

describe('[EditText]', () => {
  let value = '';

  describe('interactions', () => {
    const props = {
      testID: 'INPUT_TEST',
      testError: 'ERROR_TEST',
      type: 'box',
      borderWidth: 3,
      textStyle: {},
      labelWidth: 90,
      borderStyle: { },
      borderColor: '#fff',
      errorColor: '#fff',
      onChangeText: (word: string): void => {
        value = word;
      },
      leftElement: <View />,
      rightElement: <View />,
      rightElementStyle: {},
      leftElementStyle: {},
      focusColor: '#fff',
      inputContainerRadius: 30,
    };

    beforeEach(() => {
      component = <EditText {...props} autoCapitalize="words" />;
      testingLib = render(component);
    });

    it('should set error message when no valid email has been written', async () => {
      const input = await waitForElement(() => testingLib.getByTestId('INPUT_TEST'));
      act(() => {
        fireEvent.changeText(input, 'input test');
      });

      act(() => wait());

      expect(value).toEqual('input test');
    });

    it('should trigger onSubmit', async () => {
      const input = await waitForElement(() => testingLib.getByTestId('INPUT_TEST'));

      act(() => {
        fireEvent.submitEditing(input);
      });
    });
  });

  // errorText == undefined ? true : false
  describe('Type: [default]', () => {
    const props = {
      testID: 'INPUT_TEST',
      testError: 'ERROR_TEST',
      label: 'label text',
      errorText: undefined,
      onFocus: undefined,
      onBlur: undefined,
    };

    it('renders without crashing', () => {
      component = <EditText {...props} />;
      testingLib = render(component);
      const rendered = renderer.create(component).toJSON();
      expect(rendered).toMatchSnapshot();
      expect(rendered).toBeTruthy();
    });

    it('renders editText and running onFocus', async () => {
      props.onFocus = (): void => {};
      component = <EditText {...props} />;
      testingLib = render(component);
      const input = await waitForElement(() => testingLib.getByTestId('INPUT_TEST'));

      act(() => {
        fireEvent.focus(input);
      });
    });

    describe('when running onBlur (focused === false)', () => {
      props.onBlur = (): void => {};

      it('should trigger blur without errorText', async () => {
        component = <EditText {...props} />;
        testingLib = render(component);
        const input = await waitForElement(() => testingLib.getByTestId('INPUT_TEST'));

        act(() => {
          fireEvent.blur(input);
        });
      });

      it('should trigger blur with errorText', async () => {
        props.errorText = 'error text';
        component = <EditText {...props} />;
        testingLib = render(component);
        const input = await waitForElement(() => testingLib.getByTestId('INPUT_TEST'));

        act(() => {
          fireEvent.blur(input);
        });
      });
    });
  });

  describe('Type: [row]', () => {
    const props = {
      testID: 'INPUT_TEST',
      testError: 'ERROR_TEST',
      type: 'row',
      label: undefined,
      errorText: undefined,
      onFocus: undefined,
      onBlur: undefined,
    };

    it('renders without crashing', () => {
      component = <EditText {...props} />;
      testingLib = render(component);
      const rendered = renderer.create(component).toJSON();
      expect(rendered).toMatchSnapshot();
      expect(rendered).toBeTruthy();
    });

    it('renders row type input and runs onFocus', async () => {
      props.label = 'label text';
      props.onFocus = (): void => {};
      component = <EditText {...props} />;
      testingLib = render(component);
      const input = await waitForElement(() => testingLib.getByTestId('INPUT_TEST'));

      act(() => {
        fireEvent.focus(input);
      });
    });

    describe('when running onBlur (focused === false)', () => {
      props.onBlur = (): void => {};
      // props.errorText = 'error test';

      it('renders row type input without errorText', async () => {
        component = <EditText {...props} />;
        testingLib = render(component);
        const input = await waitForElement(() => testingLib.getByTestId('INPUT_TEST'));

        act(() => {
          fireEvent.blur(input);
        });
      });

      it('renders row type input with props errorText', async () => {
        props.errorText = 'error text';
        component = <EditText {...props} />;
        testingLib = render(component);
        const input = await waitForElement(() => testingLib.getByTestId('INPUT_TEST'));

        act(() => {
          fireEvent.blur(input);
        });
      });

      it('should trigger blur focus when onBlur and onFocus are undefined', async () => {
        props.onBlur = undefined;
        props.onFocus = undefined;
        component = <EditText {...props} />;
        testingLib = render(component);
        const input = await waitForElement(() => testingLib.getByTestId('INPUT_TEST'));

        act(() => {
          fireEvent.blur(input);
        });

        act(() => {
          fireEvent.focus(input);
        });
      });
    });
  });

  describe('Type: [box]', () => {
    const props = {
      testID: 'INPUT_TEST',
      testError: 'ERROR_TEST',
      type: 'box',
      label: 'label text',
      errorText: undefined,
      onFocus: undefined,
      onBlur: undefined,
    };

    it('renders without crashing', () => {
      component = <EditText {...props} />;
      testingLib = render(component);
      const rendered = renderer.create(component).toJSON();
      expect(rendered).toMatchSnapshot();
      expect(rendered).toBeTruthy();
    });

    it('renders editText and running onFocus', async () => {
      props.onFocus = (): void => {};
      component = <EditText {...props} />;
      testingLib = render(component);
      const input = await waitForElement(() => testingLib.getByTestId('INPUT_TEST'));

      act(() => {
        fireEvent.focus(input);
      });
    });

    describe('when running onBlur (focused === false)', () => {
      props.onBlur = (): void => {};

      it('should trigger blur without errorText', async () => {
        component = <EditText {...props} />;
        testingLib = render(component);
        const input = await waitForElement(() => testingLib.getByTestId('INPUT_TEST'));

        act(() => {
          fireEvent.blur(input);
        });
      });

      it('should trigger blur with errorText', async () => {
        props.errorText = 'error text';
        component = <EditText {...props} />;
        testingLib = render(component);
        const input = await waitForElement(() => testingLib.getByTestId('INPUT_TEST'));

        act(() => {
          fireEvent.blur(input);
        });
      });
    });
  });

  describe('Type: [rowBox]', () => {
    const props = {
      testID: 'INPUT_TEST',
      testError: 'ERROR_TEST',
      type: 'rowBox',
      label: undefined,
      errorText: undefined,
      onFocus: undefined,
      onBlur: undefined,
    };

    it('renders without crashing', () => {
      component = <EditText {...props} />;
      testingLib = render(component);
      const rendered = renderer.create(component).toJSON();
      expect(rendered).toMatchSnapshot();
      expect(rendered).toBeTruthy();
    });

    it('renders row type input and runs onFocus', async () => {
      props.label = 'label text';
      props.onFocus = (): void => {};
      component = <EditText {...props} />;
      testingLib = render(component);
      const input = await waitForElement(() => testingLib.getByTestId('INPUT_TEST'));

      act(() => {
        fireEvent.focus(input);
      });
    });

    describe('when running onBlur (focused === false)', () => {
      props.onBlur = (): void => {};
      // props.errorText = 'error test';

      it('renders row type input without errorText', async () => {
        component = <EditText {...props} />;
        testingLib = render(component);
        const input = await waitForElement(() => testingLib.getByTestId('INPUT_TEST'));

        act(() => {
          fireEvent.blur(input);
        });
      });

      it('renders row type input with props errorText', async () => {
        props.errorText = 'error text';
        component = <EditText {...props} />;
        testingLib = render(component);
        const input = await waitForElement(() => testingLib.getByTestId('INPUT_TEST'));

        act(() => {
          fireEvent.blur(input);
        });
      });

      it('should trigger blur focus when onBlur and onFocus are undefined', async () => {
        props.onBlur = undefined;
        props.onFocus = undefined;
        component = <EditText {...props} />;
        testingLib = render(component);
        const input = await waitForElement(() => testingLib.getByTestId('INPUT_TEST'));

        act(() => {
          fireEvent.blur(input);
        });

        act(() => {
          fireEvent.focus(input);
        });
      });
    });
  });
});
