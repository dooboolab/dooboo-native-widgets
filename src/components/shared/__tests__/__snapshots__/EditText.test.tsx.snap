// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[EditText] renders without crashing 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "flexDirection": "column",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    onBlur={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "color": "#2C374E",
          "fontSize": 15,
          "fontWeight": "500",
          "paddingBottom": 10,
          "paddingTop": 15,
        },
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
  />
  <View
    style={
      Array [
        Object {
          "borderColor": "#eaeaf9",
          "borderWidth": 0.6,
        },
        null,
      ]
    }
  />
</View>
`;
